packages/data-broker/src/services/dhis/DhisService.js

```diff
@@ -138,11 +138,13 @@ export class DhisService extends Service {
 
     const entityCodes = organisationUnitCodes || [organisationUnitCode];
     const pullData = this.pullers[type];
-    const apis = dataServices.map(({ isDataRegional }) =>
-      getDhisApiInstance({ entityCodes, isDataRegional }, this.models),
-    );
 
-    return pullData(apis, dataSources, options);
+    const apis = new Set();
+    dataServices.forEach(({ isDataRegional }) => {
+      apis.add(getDhisApiInstance({ entityCodes, isDataRegional }, this.models));
+    });
```

Comment:
Feels a tad risky just to rely on object reference to ensure we're not doubling up here (usually I prefer to have some kind of id or something here). I suppose that we could implement a `getServerName()` function on the `dhisApi` and key by that here? But that probably feels a bit much... and is a it bit of a duplication of logic as well...

Am cool with this as is, just wanted tor raise the consideration 👍

Reply:
Agreed. To cover this I thought I'd add some tests, will submit the PR soon